src/libs/OptionsListUtils.ts

```diff
@@ -1526,6 +1537,69 @@ function orderOptions(options: ReportUtils.OptionData[], searchValue: string | u
     );
 }
 
+function getUserToInviteOption({
+    searchValue,
+    excludeUnknownUsers = false,
+    optionsToExclude = [],
+    selectedOptions = [],
+    betas,
+    reportActions = {},
+    showChatPreviewLine = false,
+    shouldGetOptionToInvite,
+}: GetUserToInviteConfig): ReportUtils.OptionData | null {
+    let userToInvite: ReportUtils.OptionData | null = null;
+    const parsedPhoneNumber = PhoneNumber.parsePhoneNumber(LoginUtils.appendCountryCode(Str.removeSMSDomain(searchValue)));
+
+    /**
+     * We create a new user option if the following conditions are satisfied:
+     * - there's no match recent report and personal detail option
+     * - The searchValue is a valid email or phone number
+     * - The searchValue isn't the current personal detail login
+     * - We can use chronos or the search value is not the chronos email
+     */
+    if (
+        searchValue &&
+        shouldGetOptionToInvite &&
+        !isCurrentUser({login: searchValue} as PersonalDetails) &&
+        selectedOptions.every((option) => 'login' in option && option.login !== searchValue) &&
+        ((Str.isValidEmail(searchValue) && !Str.isDomainEmail(searchValue) && !Str.endsWith(searchValue, CONST.SMS.DOMAIN)) ||
+            (parsedPhoneNumber.possible && Str.isValidE164Phone(LoginUtils.getPhoneNumberWithoutSpecialChars(parsedPhoneNumber.number?.input ?? '')))) &&
+        !optionsToExclude.find((optionToExclude) => 'login' in optionToExclude && optionToExclude.login === PhoneNumber.addSMSDomainIfPhoneNumber(searchValue).toLowerCase()) &&
+        (searchValue !== CONST.EMAIL.CHRONOS || Permissions.canUseChronos(betas)) &&
+        !excludeUnknownUsers
```

Comment:
Could we clean this up a bit more? This condition is hard to read as is. And we could avoid the comment block entirely by creating some variables that are a bit easier to parse.

Reply:
@marcaaron I updated, please help to check again.